zookeeper-server/src/main/java/org/apache/zookeeper/audit/ZKAuditProvider.java

```diff
@@ -119,7 +121,9 @@ public static void addServerStartFailureAuditLog() {
     }
 
     private static void log(String user, String operation, Result result) {
-        auditLogger.logAuditEvent(createLogEvent(user, operation, result));
+        if (isAuditEnabled()) {
+            auditLogger.logAuditEvent(createLogEvent(user, operation, result));
+        }
     }
```

Comment:
This method is the same, when `'AUDIT_ENABLE'` is false, the program cannot run at here, so no NPE issue. Am I missing something?